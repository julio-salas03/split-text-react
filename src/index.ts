export * from './components/index';
export type {
  LetterWrapperProps,
  LineWrapperProps,
  WordWrapperProps,
} from './components/Wrappers';
