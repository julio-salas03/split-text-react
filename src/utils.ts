export function debounce<T extends any[]>(
  func: (...args: T) => any,
  timeout: number
): (...args: T) => void {
  let timer: NodeJS.Timeout;
  return (...args: T) => {
    clearTimeout(timer);
    timer = setTimeout(() => {
      func(...args);
    }, timeout);
  };
}
