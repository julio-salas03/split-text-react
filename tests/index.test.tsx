import React from 'react';
import { render } from '@testing-library/react';
import { SplitText } from '../src/index';
import ResizeObserver from 'resize-observer-polyfill';

describe('SplitText', () => {
  beforeAll(() => {
    global.ResizeObserver = ResizeObserver;
  });

  beforeAll(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {});
  });

  it('should only accept text children', () => {
    expect(() =>
      render(
        <SplitText>
          <div>foo</div>
        </SplitText>
      )
    ).toThrow();
  });

  it('should support computed text', () => {
    expect(() => render(<SplitText>foo {5}</SplitText>)).not.toThrow();
  });

  it('should forward the ref to the component', () => {
    const Component: React.FC = () => {
      const ref = React.useRef(null);
      setTimeout(() => {
        expect(ref.current).not.toBeNull();
        expect(ref.current).toBeInstanceOf(HTMLElement);
      }, 500);
      return <SplitText ref={ref}>Hello World</SplitText>;
    };

    render(<Component />);
  });
});
