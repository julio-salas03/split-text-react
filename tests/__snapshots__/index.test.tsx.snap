// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitText should break text as in the captured snapshot 1`] = `
<div>
  <span
    style="display: block;"
  >
    <span
      style="display: block;"
    >
      <span
        style="white-space: pre;"
      >
        <span>
          L
        </span>
        <span>
          o
        </span>
        <span>
          r
        </span>
        <span>
          e
        </span>
        <span>
          m
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          i
        </span>
        <span>
          p
        </span>
        <span>
          s
        </span>
        <span>
          u
        </span>
        <span>
          m
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          d
        </span>
        <span>
          o
        </span>
        <span>
          l
        </span>
        <span>
          o
        </span>
        <span>
          r
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          s
        </span>
        <span>
          i
        </span>
        <span>
          t
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          a
        </span>
        <span>
          m
        </span>
        <span>
          e
        </span>
        <span>
          t
        </span>
        <span>
          ,
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          c
        </span>
        <span>
          o
        </span>
        <span>
          n
        </span>
        <span>
          s
        </span>
        <span>
          e
        </span>
        <span>
          c
        </span>
        <span>
          t
        </span>
        <span>
          e
        </span>
        <span>
          t
        </span>
        <span>
          u
        </span>
        <span>
          r
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          a
        </span>
        <span>
          d
        </span>
        <span>
          i
        </span>
        <span>
          p
        </span>
        <span>
          i
        </span>
        <span>
          s
        </span>
        <span>
          c
        </span>
        <span>
          i
        </span>
        <span>
          n
        </span>
        <span>
          g
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          e
        </span>
        <span>
          l
        </span>
        <span>
          i
        </span>
        <span>
          t
        </span>
        <span>
          .
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          E
        </span>
        <span>
          t
        </span>
        <span>
          i
        </span>
        <span>
          a
        </span>
        <span>
          m
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          a
        </span>
        <span>
          c
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          u
        </span>
        <span>
          r
        </span>
        <span>
          n
        </span>
        <span>
          a
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          h
        </span>
        <span>
          e
        </span>
        <span>
          n
        </span>
        <span>
          d
        </span>
        <span>
          r
        </span>
        <span>
          e
        </span>
        <span>
          r
        </span>
        <span>
          i
        </span>
        <span>
          t
        </span>
        <span>
          ,
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          a
        </span>
        <span>
          l
        </span>
        <span>
          i
        </span>
        <span>
          q
        </span>
        <span>
          u
        </span>
        <span>
          e
        </span>
        <span>
          t
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          q
        </span>
        <span>
          u
        </span>
        <span>
          a
        </span>
        <span>
          m
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          e
        </span>
        <span>
          g
        </span>
        <span>
          e
        </span>
        <span>
          t
        </span>
        <span>
          ,
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          f
        </span>
        <span>
          a
        </span>
        <span>
          u
        </span>
        <span>
          c
        </span>
        <span>
          i
        </span>
        <span>
          b
        </span>
        <span>
          u
        </span>
        <span>
          s
        </span>
        <span>
           
        </span>
      </span>
      <span
        style="white-space: pre;"
      >
        <span>
          l
        </span>
        <span>
          e
        </span>
        <span>
          c
        </span>
        <span>
          t
        </span>
        <span>
          u
        </span>
        <span>
          s
        </span>
        <span>
          .
        </span>
      </span>
    </span>
  </span>
</div>
`;
